sw/host/tests/rom/e2e_bootstrap_rma/src/main.rs

```diff
@@ -18,362 +21,322 @@ use opentitanlib::dif::lc_ctrl::{
-        TransitionTarget, TransitionToken0, TransitionToken1, TransitionToken2, TransitionToken3,
+/// Iteration 1 of the `rom_bootstrap_rma` testpoint.
+///
+/// Verifies that the ROM times out on spin cycles, automatically resets the
+/// device, and logs the expected reset reasons. Does not issue the life cycle
+/// RMA command.
+fn test_no_rma_command(opts: &Opts, transport: &TransportWrapper) -> anyhow::Result<()> {
+    let uart = transport.uart("console").context("failed to get UART")?;
+
+    reset_with_strapping(
+        transport,
+        &["RMA_BOOTSTRAP"],
+        opts.init.bootstrap.options.reset_delay,
+    )
+    .context("failed to reset with RMA_BOOTSTRAP strapping")?;
+
+    log::info!("Waiting for reset reasons over the console...");
+
+    // Strings expected from the console representing success or failure.
+    let expected_bitfield = u32::from(DifRstmgrResetInfo::Por) | u32::from(DifRstmgrResetInfo::Sw);
+    let exit_success =
+        Regex::new(format!("reset_info_bitfield: 0x{expected_bitfield:x}\r\n").as_str())
+            .context("failed to build regex")?;
+    let exit_failure =
+        Regex::new("reset_info_bitfield: 0x[0-9a-f]+\r\n").context("failed to build regex")?;
+
+    let mut console = UartConsole {
+        timeout: Some(CONSOLE_TIMEOUT),
+        exit_success: Some(exit_success),
+        exit_failure: Some(exit_failure),
+        ..Default::default()
     };
-    use OpenocdTclBlock::{AssertRegEq, Echo, PollUntilRegEq, WriteReg};
 
+    let mut stdout = io::stdout();
+    let result = console
+        .interact(&*uart, None, Some(&mut stdout))
+        .context("failed to interact with console")?;
+
+    match result {
+        ExitStatus::ExitSuccess => Ok(()),
+        result => bail!("FAIL: {result:?}"),
+    }
+}
+
+/// Iteration 2 of the `rom_bootstrap_rma` testpoint.
+///
+/// Issues the lifecycle RMA command and ensures the RMA transition is completed
+/// successfully.
+fn test_rma_command(opts: &Opts, transport: &TransportWrapper) -> anyhow::Result<()> {
+    let rma_bootstrap_strapping = transport.pin_strapping("RMA_BOOTSTRAP")?;
+    let tap_strapping = transport.pin_strapping("PINMUX_TAP_LC")?;
+
+    // Need to reset with `RMA_BOOTSTRAP` set.
+    rma_bootstrap_strapping
+        .apply()
+        .context("failed to apply RMA_BOOTSTRAP strapping")?;
+
+    // We keep `PINMUX_TAP_LC` set so that resets due to transition don't
+    // disconnect us from the LC TAP.
+    tap_strapping
+        .apply()
+        .context("failed to apply PINMUX_TAP_LC strapping")?;
+
+    transport
+        .reset_target(opts.init.bootstrap.options.reset_delay, true)
+        .context("failed to reset target")?;
+
+    // Remove the RMA_BOOTSTRAP strapping to bring the device out of the RMA spin loop.
+    rma_bootstrap_strapping
+        .remove()
+        .context("failed to remove strapping RMA_BOOTSTRAP")?;
+
+    // NOTE: without this sleep, the transition starts but isn't successful.
```

Comment:
That is peculiar. Are you saying that when you go to issue an LC transition, and you don't have this wait, you read back a 0x3 from the status reg, which decodes to INITIALIZED and READY bits set, yet you still can't perform a transition?